source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/locale-input/field.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+import { useMemo, memo } from 'react';
+import Autosuggest from 'react-autosuggest';
+import { assert } from '@orbit/utils';
+
+import { withTranslations, i18nProps, useLdml } from '~/lib/i18n';
+
+import { filterForValidAttributes } from '~/lib/dom';
+
+import { getSuggestions, findLanguageCode, sortComparer, findLanguageInfo } from './-utils/helpers';
+import { localizeTagData } from './-utils/localize';
+import { Suggestion } from './suggestion';
+
+export interface IProps {
+  value: string; // tag
+  onChange: (localeCode: string) => void;
+  data: ILanguageInfo[];
+}
+
+interface IState {
+  value?: string;
+  isMatch: boolean;
+  suggestions: ILanguageInfo[];
+  languageInfo: ILanguageInfo;
+}
+
+class Field extends React.Component<IProps & i18nProps, IState> {
+  getSuggestions: (value: string) => ILanguageInfo[];
+  findLanguageCode: (value: string) => string;
+  findLanguageInfo: (value: string) => ILanguageInfo;
+
+  constructor(props) {
+    super(props);
+
+    assert(`The Locale Picker Field needs a data set. The passed data set is empty.`, props.data);
+
+    // TODO: re-calc this if the user changes their language
```

Comment:
I have a plan to convert this to react-hooks later, which will allow me to memoize a specific property within the function, which will allow updates when the currently selected language changes.